src/Microsoft.SqlTools.ServiceLayer/BatchParser/BatchParserWrapper.cs

```diff
@@ -111,6 +117,73 @@ public sealed class BatchParserWrapper : IDisposable
             return batchDefinitionList;
         }
 
+        private static int GetMaxStartLine(IList<Tuple<int, int>> positions) 
+        {
+            int highest = 0;
+            foreach (var position in positions) 
+            {
+                if (position.Item1 > highest) 
+                {
+                    highest = position.Item1;
+                }
+            }
+            return highest;
+        }
+
+        /// <summary>
+        /// Gets offsets for all batches
+        /// </summary>
+        private static List<int> GetOffsets(string content, IList<Tuple<int, int>> positions) 
+        {
+
+            List<int> offsets = new List<int>();
+            int count = 0;
+            int offset = 0;
+            bool foundAllOffsets = false;
+            using (StringReader reader = new StringReader(content)) {
+                while (!foundAllOffsets)
+                 {
+                    for (int i = 0; i <= GetMaxStartLine(positions); i++)
```

Comment:
GetMaxStartLine is being called within a loop with the same parameter.  Should you call it once outside the loop and cache the return value?